### ISSUE TITLE
Unexpected attribute error on executing graph with introspection

### ISSUE THREAD

**Issue by playpauseandstop** [vmagamedov/hiku#15]
Long story short: after wrapping high level graph into `add_introspection_async` I'm unable to execute any query to it due:

```
Traceback (most recent call last):
  File "/work/app/graph/views.py", line 41, in query_graph
    result = await engine.execute(graph, query, context)
  File "/usr/local/lib/python3.6/dist-packages/hiku/executors/asyncio.py", line 17, in process
    queue.progress(done)
  File "/usr/local/lib/python3.6/dist-packages/hiku/executors/queue.py", line 38, in progress
    callback()
  File "/usr/local/lib/python3.6/dist-packages/hiku/engine.py", line 217, in <lambda>
    self.process_link(node, _gl, _qe, ids, _fut.result())
  File "/usr/local/lib/python3.6/dist-packages/hiku/engine.py", line 241, in process_link
    to_ids)
  File "/usr/local/lib/python3.6/dist-packages/hiku/engine.py", line 180, in process_node
    node, func_fields, ids)
  File "/usr/local/lib/python3.6/dist-packages/hiku/sources/graph.py", line 91, in __call__
    for e in graph_fields)}
  File "/usr/local/lib/python3.6/dist-packages/hiku/sources/graph.py", line 89, in <dictcomp>
    fn_env = {f.__def_name__: f.__def_body__
  File "/usr/local/lib/python3.6/dist-packages/hiku/sources/graph.py", line 91, in <genexpr>
    for e in graph_fields)}
AttributeError: 'Field' object has no attribute 'functions'
```

----

The engine & graph enabled as,

```py
async def init_hiku(app: web.Application) -> None:
    app['hiku_engine'] = HikuEngine(AsyncIOExecutor(app.loop))
    app['hiku_graph'] = add_introspection_async(HIGH_LEVEL_GRAPH)
```

----

GraphQL aiohttp handler,

```py
@login_required
async def query_graph(request: web.Request, user: RowProxy) -> web.Response:
    engine, graph = request.app['hiku_engine'], request.app['hiku_graph']
    data = validate_query_graph(await request.json())

    try:
        # Prepare Hiku query
        query = read(data['query'], data['variables'])

        # Validate query
        validator = QueryValidator(graph)
        validator.visit(query)
        if validator.errors.list:
            return graphql_response(errors=validator.errors.list)

        # Execute graph
        context = {
            DB_ENGINE_KEY: request.app['db'],
            SESSION_KEY: await get_session(request),
            USER_KEY: user,
        }
        result = await engine.execute(graph, query, context)

        # Denormalize results
        graph_data = denormalize(graph, result, query)
    except Exception as err:
        logger.exception(
            'Unhandled exception on executing GraphQL',
            extra=data)
        return graphql_response(errors=[str(err)])

    return graphql_response(data=graph_data)
```

----

I attempted to execute _any_ GraphQL query, but all of them ends with same error. Let me know which else information you needed to reproduce the error. The high level graph is quite large to be copy-pasted here.

**Comment by vmagamedov** [comment]
Can confirm that introspection is not working with high-level graphs, will fix this soon.

**Comment by vmagamedov** [comment]
Fixed in a13b9ff55d5e25f93cb28ce8e90e4b28eb9abea5 (`hiku==0.3.9rc2`), can you confirm and close this issue?

**Comment by playpauseandstop** [comment]
After update to `0.3.9rc2` executing GraphQL queries results in,

```
 16:33:16 ERROR app.graph.views v7.0.1-2-gd22aaaa2 00000000-0000-0000-0000-000000000000: Unhandled exception on executing GraphQL
Traceback (most recent call last):
  File "/work/app/graph/views.py", line 31, in query_graph
    validator.visit(query)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 146, in visit
    return obj.accept(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 117, in accept
    return visitor.visit_node(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 156, in visit_node
    self.visit(item)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 146, in visit
    return obj.accept(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 97, in accept
    return visitor.visit_link(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/validate/query.py", line 199, in visit_link
    self.visit(obj.node)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 146, in visit
    return obj.accept(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 117, in accept
    return visitor.visit_node(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 156, in visit_node
    self.visit(item)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 146, in visit
    return obj.accept(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/query.py", line 76, in accept
    return visitor.visit_field(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/validate/query.py", line 166, in visit_field
    is_field = _AssumeField(node, self.errors).visit(field)
  File "/usr/local/lib/python3.6/dist-packages/hiku/graph.py", line 409, in visit
    return obj.accept(self)
  File "/usr/local/lib/python3.6/dist-packages/hiku/sources/graph.py", line 66, in accept
    return visitor.visit_expr(self)
AttributeError: '_AssumeField' object has no attribute 'visit_expr'
```

**Comment by playpauseandstop** [comment]
Also introspection still works, but validation non-introspection queries fails with error above

**Comment by vmagamedov** [comment]
Fixed this issue in 3c7c9f25b26d5fb8a960e0525180be8780b0bd76, `hiku==0.3.9rc3`

**Comment by playpauseandstop** [comment]
After bumping to `0.3.9rc3` executing works as well as introspection. Closing this as `fixed`
